src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectPropertiesProviders/ImplicitProjectPropertiesProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.ProjectPropertiesProviders
+{
+    /// <summary>
+    /// Provides project properties that normally should not live in the project
+    /// file but may be written from an external source (e.g. the solution file).
+    /// This provider avoids writing these properties to the project file, but if
+    /// they are already present there, it updates the value to keep in sync with
+    /// the external source.
+    /// </summary>
+    [Export("Implicit", typeof(IProjectPropertiesProvider))]
+    [Export(typeof(IProjectPropertiesProvider))]
+    [ExportMetadata("Name", "Implicit")]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class ImplicitProjectPropertiesProvider : DelegatedProjectPropertiesProviderBase
+    {
+        [ImportingConstructor]
+        public ImplicitProjectPropertiesProvider(
+            [Import("Microsoft.VisualStudio.ProjectSystem.ProjectFile")] IProjectPropertiesProvider provider)
```

Comment:
> if we move this to the basetype, then all subtypes are delegating to the ProjectFilePropertiesProvider 

Do we know of a case where we will need to delegate to any other default provider? I am currently writing a project properties provider that just does some validation/modification to a string property and I was going to derive from DelegatedProjectPropertiesProviderBase, and would need to unnecessarily repeat the same constructor.